From 28d590900ab20dec3dc447562aefaa5e2771c48e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Tue, 28 Feb 2017 13:58:20 +0100
Subject: [PATCH 86/90] secrets: support HTTP basic authentication with proxy
 provider

Even though configuration options auth_type = basic, username and password
are read they were not used anywhere prior this patch.

Reviewed-by: Simo Sorce <simo@redhat.com>
Reviewed-by: Jakub Hrozek <jhrozek@redhat.com>
(cherry picked from commit af026ea6a6e812b7d6c5c889dda64ba7b7c433ee)
---
 src/responder/secrets/proxy.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/responder/secrets/proxy.c b/src/responder/secrets/proxy.c
index 240a1de1e431d511a1eca24d8b463c37ba893e7b..fd96e985c897e2cb470a9b5d6eecbd34350fb7d2 100644
--- a/src/responder/secrets/proxy.c
+++ b/src/responder/secrets/proxy.c
@@ -440,6 +440,15 @@ static errno_t proxy_http_create_request(TALLOC_CTX *mem_ctx,
         }
     }
 
+    /* Set basic authentication if required. */
+    if (pcfg->auth_type == PAT_BASIC_AUTH) {
+        ret = tcurl_req_http_basic_auth(tcurl_req, pcfg->auth.basic.username,
+                                        pcfg->auth.basic.password);
+        if (ret != EOK) {
+            goto done;
+        }
+    }
+
     talloc_steal(tcurl_req, body);
     *_tcurl_req = talloc_steal(mem_ctx, tcurl_req);
 
-- 
2.9.3

